src/crate/client/sqlalchemy/types.py

```diff
@@ -131,24 +132,31 @@ def __eq__(self, other):
         return dict.__eq__(self, other)
 
 
-class _Craty(sqltypes.UserDefinedType):
+class ObjectTypeImpl(sqltypes.UserDefinedType, sqltypes.JSON):
+
+    __visit_name__ = "OBJECT"
+
     cache_ok = False
+    none_as_null = False
 
-    class Comparator(sqltypes.TypeEngine.Comparator):
 
-        def __getitem__(self, key):
-            return default_comparator._binary_operate(self.expr,
-                                                      operators.getitem,
-                                                      key)
+# Designated name to refer to. `Object` is too ambiguous.
+ObjectType = MutableDict.as_mutable(ObjectTypeImpl)
```

Comment:
SQLAlchemy also offers a `MutableDict` implementation. It looks like using that would resolve some other specializations of the CrateDB dialect, which are a bit anomalous, as we discovered on behalf of the discussion at https://github.com/sqlalchemy/sqlalchemy/discussions/5915.

However, it did not work out of the box, SQLAlchemy's `MutableDict`, or associated code paths, did not seem to support _nested_ object access well. Maybe indeed this is one of the essential specializations the CrateDB dialect has to offer.

This topic will need to be re-visited. /cc GH-425